orte/mca/oob/tcp/oob_tcp_common.c

```diff
@@ -100,14 +96,88 @@ void orte_oob_tcp_set_socket_options(int sd)
     }
 #endif
 #if defined(SO_RCVBUF)
-    if(mca_oob_tcp_component.tcp_rcvbuf > 0 &&
-       setsockopt(sd, SOL_SOCKET, SO_RCVBUF, (char *)&mca_oob_tcp_component.tcp_rcvbuf, sizeof(int)) < 0) {
+    if (mca_oob_tcp_component.tcp_rcvbuf > 0 &&
+        setsockopt(sd, SOL_SOCKET, SO_RCVBUF, (char *)&mca_oob_tcp_component.tcp_rcvbuf, sizeof(int)) < 0) {
         opal_output(0, "[%s:%d] setsockopt(SO_RCVBUF) failed: %s (%d)", 
                     __FILE__, __LINE__, 
                     strerror(opal_socket_errno),
                     opal_socket_errno);
     }
 #endif
+#if defined(SO_KEEPALIVE)
+    if (0 < mca_oob_tcp_component.keepalive_time) {
+        int option;
+        socklen_t optlen;
+        
+        /* see if the keepalive option is available */
+        optlen = sizeof(option);
+        if (getsockopt(sd, SOL_SOCKET, SO_KEEPALIVE, &option, &optlen) < 0) {
+            /* not available, so just return */
+            return;
```

Comment:
do we really want to `return` in all these error cases?  It might be best to put all the keepalive logic in one function that we call from `orte_oob_tcp_set_socket_options()` so that you can still just return from that function but a future dev who adds new socket options at the bottom of `orte_oob_tcp_set_socket_options` doesn't hit problems due to keepalive error handling.